python-lib/nlp_utils.py

```diff
@@ -3,16 +3,45 @@
 import unicodedata
 
 
-def get_keyword(text: AnyStr, normalize_case: bool) -> AnyStr:
-    """Return text in its wanted-case form"""
-    return text.lower() if normalize_case else text
+def normalize_case_text(text: AnyStr, lowercase: bool) -> AnyStr:
+    """Return text in its wanted case form"""
+    return text.lower() if lowercase else text
 
 
-def get_sentence(span: Span, normalize_case: bool) -> Union[Span, Doc]:
-    """Return Span object as a Doc if case_insensitive is set to True"""
-    return span if normalize_case else span.as_doc()
+def get_token_attribute(lemmatize: bool) -> AnyStr:
+    """Return the attribute to pass to spaCy Matcher"""
+    return "LEMMA" if lemmatize else "TEXT"
 
 
-def normalize_text(texts: List[AnyStr]) -> List[AnyStr]:
+def lemmatize_doc(doc: Doc) -> AnyStr:
+    """Lemmatize SpaCy.tokens.Doc object.
+
+    Args:
+        doc (SpaCy.tokens.Doc): Text to lemmatize
+
+    Returns:
+        str: Text in its lemmatized form
+
+    """
+    return " ".join([span.lemma_ for span in doc])
+
+
+def lemmatize_span(span: Span, lemmatize: bool) -> AnyStr:
```

Comment:
A bit weird to have a `lemmatize` argument in a `lemmatize` function that says if the function will lemmatize or not. Maybe give the function a more general name ?
```suggestion
def get_span_text(span: Span, lemmatize: bool) -> AnyStr:
```

Reply:
Good idea! I did it in this commit 8b8549a